controllers/storagecluster/cephblockpools.go

```diff
@@ -156,7 +156,18 @@ func (obj *ocsCephBlockPools) ensureCreated(r *StorageClusterReconciler, instanc
 
 			r.Log.Info("Restoring original CephBlockPool.", "CephBlockPool", klog.KRef(cephBlockPool.Namespace, cephBlockPool.Name))
 			existing.ObjectMeta.OwnerReferences = cephBlockPool.ObjectMeta.OwnerReferences
-			existing.Spec = cephBlockPool.Spec
+			existing.Spec.DeviceClass = cephBlockPool.Spec.DeviceClass
+			existing.Spec.FailureDomain = cephBlockPool.Spec.FailureDomain
+			existing.Spec.Replicated = cephBlockPool.Spec.Replicated
+			existing.Spec.EnableRBDStats = cephBlockPool.Spec.EnableRBDStats
+			// update the mirroring field only when it is not nil
+			if instance.Spec.Mirroring != nil {
+				existing.Spec.Mirroring = cephBlockPool.Spec.Mirroring
+			}
+			// update the name only if it is NFS blockpool
+			if cephBlockPool.Name == generateNameForCephNFSBlockPool(instance) && instance.Spec.NFS != nil && instance.Spec.NFS.Enable {
+				existing.Spec.Name = cephBlockPool.Spec.Name
+			}
```

Comment:
if `existing.Spec` is being replaced by all these conditional entries, what happens if some new field gets added to `cephBlockPool` and we failed to update that in here?

but again, this is only true if the initial assumption is, this piece doesn't require an update if rook api gets updated.